rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -120,12 +121,30 @@ final class ParameterPosition extends TParameterPosition {
   }
 }
 
+/** Holds if `call` invokes a qualified path that resolves to a method. */
+private predicate callToMethod(CallExpr call) {
+  exists(Path path |
+    path = call.getFunction().(PathExpr).getPath() and
+    path.hasQualifier() and
```

Comment:
Why is this restriction needed?

Reply:
As far as I can tell, a method can only be called as a function by including a qualifying path. So if the path doesn't have a qualifier, then it will never resolve to a method. So I'm including it for clarity and (maybe) performance.

Reply:
Yeah, I think you are right.